### PR TITLE
Automate operator release

### PR DESCRIPTION
**Pull request by agatakrajewska**
- [ ] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Automate raising a PR from our fork to operator-framework/community-operators.
As a first step we're adding a python script, which we later can incorporate into the pipeline.

### Notes for the reviewer

Commit by commit

### More information

- [Jira ticket RUN-1058](https://snyksec.atlassian.net/browse/RUN-1058)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -312,7 +312,7 @@ jobs:
         command: |
           set -xeo pipefail
 
-          OPERATOR_VERSION=$(python ./scripts/operator/get-last-published-operator-version.py)
+          OPERATOR_VERSION=$(python ./scripts/operator/get_last_published_operator_version.py)
```
Comment (agranado2k) [inline .circleci/config.yml:315]: I think, after @ivanstanev has split the `config.yml` in different files, we can't change it directly. Instead you should make changes in the files on the folder `.circleci/config` and run this commands [here](https://github.com/snyk/kubernetes-monitor/blob/staging/.circleci/README.md) to re-generate the `config.yml`.

File: scripts/operator/pr_resources.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+
+content = ''' Signed-off-by: Ivan Stanev istanev93@gmail.com
```
Comment (agranado2k) [inline scripts/operator/pr_resources.py:3]: Good stuff! 😁

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    for refs in repo.refs:
+        if 'snyk-operator' in refs.name:
+            operator_branches.append(refs.name)
+
+    return operator_branches[-1].split('origin/')[-1]
+
+
+def create_pull_request(new_operator_version, new_release_branch_name):
+    github = Github('23ba6faa9635ff064f7bf6a6c36a0a33dff4f376')
```
Comment (agranado2k) [inline scripts/operator/raise_pr_to_community_operators_from_our_fork.py:25]: Should we put this id in an env var (and in 1Password to make it easier to get it for development)?

File: .circleci/config.yml [hunk 4]
```diff
@@ -906,4 +906,5 @@ workflows:
             - master
         requires:
         - build_image
+    - operator_upgrade_tests
```
Comment (agranado2k) [inline .circleci/config.yml:909]: Same comment that I put in a previous commit.
I think, after @ivanstanev has split the config.yml in different files, we can't change it directly. Instead you should make changes in the files on the folder .circleci/config and run this commands here to re-generate the config.yml.

File: src/scanner/images/index.ts [hunk 5]
```diff
@@ -41,18 +41,31 @@ export async function removePulledImages(images: IPullableImage[]): Promise<void
 }
 
 // Exported for testing
-export function removeTagFromImage(imageWithTag: string): string {
-  return imageWithTag.split('@')[0].split(':')[0];
+export function getImageParts(imageWithTag: string) : {imageName: string, imageTag: string} {
```
Comment (agranado2k) [inline src/scanner/images/index.ts:44]: Should this commit be in this PR? 🤔

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    repo = Repo.clone_from(
+        'https://github.com/snyk/community-operators.git', './community-operators-repo')
+
+    operator_branches = []
+
+    for refs in repo.refs:
+        if 'snyk-operator' in refs.name:
+            operator_branches.append(refs.name)
+
+    return operator_branches[-1].split('origin/')[-1]
```
Comment (shaimendel) [inline scripts/operator/raise_pr_to_community_operators_from_our_fork.py:21]: are we sure this logic is correct? how is it ordered?

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    github = Github('access_token')
+    user = github.get_user('operator-framework')
+    repo = user.get_repo('community-operators')
+    body = pr_resources.content
+
+    pr = repo.create_pull(title='Upgrade snyk-operator to version ' + new_operator_version,
+                          body=body,
+                          base='master',
+                          head='snyk:' + new_release_branch_name,
+                          draft=True)
```
Comment (shaimendel) [inline scripts/operator/raise_pr_to_community_operators_from_our_fork.py:34]: should the draft=true be deleted?

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import os
+from git import Repo
+from github import Github
+import shutil
+from get_last_published_operator_version import getLastOperatorVersion
+import pr_resources
+
+
+def get_latest_operator_branch():
+    repo = Repo.clone_from(
```
Comment (shaimendel) [inline scripts/operator/raise_pr_to_community_operators_from_our_fork.py:12]: isn't there a way to get the list of refs without cloning it? can't you get it from the Github package?

File: .circleci/config.yml (hunk 1 above)
Comment (agatakrajewska) [reply .circleci/config.yml]: Ah of course, this is the generated file 🤦‍♀️ 👍

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py (hunk 3 above)
Comment (agatakrajewska) [reply scripts/operator/raise_pr_to_community_operators_from_our_fork.py]: Yes, I'll add an env var

File: .circleci/config.yml (hunk 4 above)
Comment (agatakrajewska) [reply .circleci/config.yml]: 👍

File: src/scanner/images/index.ts (hunk 5 above)
Comment (agatakrajewska) [reply src/scanner/images/index.ts]: rebased 👍

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py (hunk 7 above)
Comment (agatakrajewska) [reply scripts/operator/raise_pr_to_community_operators_from_our_fork.py:34]: yes

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py (hunk 6 above)
Comment (agatakrajewska) [reply scripts/operator/raise_pr_to_community_operators_from_our_fork.py:21]: yes, I've tested it adding a new branch, it looks like it's always last

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py (hunk 8 above)
Comment (agatakrajewska) [reply scripts/operator/raise_pr_to_community_operators_from_our_fork.py:12]: I know, that would be great! I've tried with pygit yesterday but didn't actually try with pygithub, I'll have a look!

Comment (agatakrajewska) [reply scripts/operator/raise_pr_to_community_operators_from_our_fork.py]: it was so easy :-)

File: scripts/operator/pr_resources.py (hunk 2 above)
Comment (agatakrajewska) [reply scripts/operator/pr_resources.py:3]: 😆

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+def get_latest_operator_branch():
+    repo = github.get_repo('snyk/community-operators')
+    branches = repo.get_branches()
+    operator_branches = []
+
+    for branch in branches:
+        if 'snyk-operator' in branch.name:
+            operator_branches.append(branch.name)
+
+    return operator_branches[-1]
```
Comment (shaimendel) [inline scripts/operator/raise_pr_to_community_operators_from_our_fork.py:23]: FYI - a shorter option (haven't checked the syntax :) )
```suggestion
    return [branch.name for branch in branches if 'snyk-operator' in branch.name][-1]
```

File: scripts/operator/raise_pr_to_community_operators_from_our_fork.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    new_operator_release_branch = get_latest_operator_branch()
+    print(
+        'Latest operator branch in snyk/community-operators: ' + new_operator_release_branch)
+    new_operator_version = new_operator_release_branch.split('-v')[-1]
+    print('Operator version to be released: ' + new_operator_version)
+    # if new_operator_version == latest_published_operator_version:
+    #     print('No new operator versions to publish')
+    #     os.exit(1)
+    # else:
+    #     create_pull_request(new_operator_version, new_operator_release_branch)
```
Comment (ivanstanev) [inline scripts/operator/raise_pr_to_community_operators_from_our_fork.py:50]: Should these not be commented out?

Comment (agatakrajewska) [reply scripts/operator/raise_pr_to_community_operators_from_our_fork.py]: fixed

**Pull request merged**